test/python/quantum_info/operators/test_zero_op.py

```diff
@@ -0,0 +1,506 @@
+# -*- coding: utf-8 -*-
+
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2017, 2020.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+# pylint: disable=invalid-name
+
+"""Tests for ZeroOp class."""
+
+import unittest
+import numpy as np
+
+from qiskit.test import QiskitTestCase
+from qiskit.quantum_info.operators import ZeroOp, ScalarOp, Operator
+
+
+class TestZeroOp(QiskitTestCase):
```

Comment:
These tests, as test/python/quantum_info/operators/test_scalar_op.py before 73d5cc5, seem too monolithic and the could be broken. Breaking tests in smaller cases give more information about fails when they fail.

Reply:
I thought that was the whole point of subTests: each failing subtest shows as separate failure when they fail.

Reply:
But subTests cannot run independently. Also, the subTests make easier to accidentally create dependency between tests, which is something not recommended...

Sometimes unfolding a subtest is too much work and does not increase the case coverage much (somebody can argue that that test is necessary anyway)... so I'm not saying that is totally removable in every situation.

Reply:
Fair enough